src/borg/crypto.pyx

```diff
@@ -480,23 +484,33 @@ cdef class _AEAD_BASE:
             iv_out[i] = iv[i]
 
 
-cdef class AES256_GCM(_AEAD_BASE):
+cdef class _AES_BASE(_AEAD_BASE):
+    def block_count(self, length):
+        return num_aes_blocks(length)
+
+
+cdef class _CHACHA_BASE(_AEAD_BASE):
+    def block_count(self, length):
+        return (length + 63) // 64
+
+
+cdef class AES256_GCM(_AES_BASE):
     def __init__(self, mac_key, enc_key, iv=None):
         if OPENSSL_VERSION_NUMBER < 0x10001040:
             raise ValueError('AES GCM requires OpenSSL >= 1.0.1d. Detected: OpenSSL %08x' % OPENSSL_VERSION_NUMBER)
```

Comment:
that the check is here implies that this can vary from system to system. therefore it would be good to be able to check / list what is available, which is easier if that's a static method and doesn't require trial construction of these objects.